### PR TITLE
EZP-25771: Inject block into custom-fetchers.

### PR DESCRIPTION
**Pull request by whitefire**
Link: https://jira.ez.no/browse/EZP-25771

Lets do this! :)

### CODE REVIEW

File: packages/ezflow_extension/ezextension/ezflow/classes/ezflowoperations.php [hunk 1]
```diff
@@ -73,6 +73,7 @@ public static function updateBlockPoolByBlockID( $block, $publishedBeforeOrAt =
         $fetchClassOptions->iniFile = 'block.ini';
         $fetchClassOptions->iniSection = $block['block_type'];
         $fetchClassOptions->iniVariable = 'FetchClass';
+        $fetchClassOptions->handlerParams = array(new eZFlowFetchParameters($block));
```
Comment (crevillo) [inline packages/ezflow_extension/ezextension/ezflow/classes/ezflowoperations.php]: should we follow old legacy coding standars here? If so, some spaces are missing.

Comment (bdunogier) [inline packages/ezflow_extension/ezextension/ezflow/classes/ezflowoperations.php]: Yes, we should.

**Comment by yannickroger** [comment]
The issue has been moved in the right project (EZP), can you please update the issue number ?

**Comment by whitefire** [comment]
Issue-number updated. Spaces added.

**Comment by andrerom** [comment]
+1

**Comment by crevillo** [comment]
+1

**Comment by yannickroger** [comment]
Should we backport it ?

**Comment by bdunogier** [comment]
I'd like a little extra explanation of how this is used. From the perspective of this change alone, we add a new class, without saying at any point what is used for and where from.

That aside, +1.

**Comment by whitefire** [comment]
In it's current state, all the fetcher has access to are the fetch-parameters (dynamic and fixed). By giving access to the entire block, a lot of other data also becomes implicitly available. Like the content-object that contains the block.

The specific use-case that warranted this pull-request was a so called "Chronological block". Requirements dictates that this type of block should display the latest articles, but exclude any articles already visible on the page. The fetcher thus requires knowledge of all previous blocks.

**Comment by whitefire** [comment]
Are we ready to merge, or is there something else I need to do?

File: packages/ezflow_extension/ezextension/ezflow/classes/ezflowfetchparameters.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /** @var array */
+    public $block;
+
+    public function __construct( array $block )
+    {
+        $this->block = $block;
+    }
+}
+
```
Comment (bdunogier) [inline packages/ezflow_extension/ezextension/ezflow/classes/ezflowfetchparameters.php]: Could you remove the blank line and the PHP closing tag ?

**Comment by bdunogier** [comment]
+1 with the CS detail fixed.

**Comment by whitefire** [comment]
Aaaand it's gone.

**Comment by andrerom** [comment]
@yannickroger may you do the honor of merging? :)

**Comment by yannickroger** [comment]
Thanks @whitefire

**Comment by whitefire** [comment]
Sorry for commenting here instead of on the jira-issue, but I was unable to log in. Some trouble with the service. Regarding a usage example: havard@byte.no can provide you with one.

**Comment by yannickroger** [comment]
Could you please help our QA team on the JIRA issue (or here) as they are not able to test this because of the lack of documentation.
We will have to revert the patch if they don't validate it.

**Comment by andrerom** [comment]
regression: https://github.com/ezsystems/ezpublish-legacy/pull/1246

**Pull request merged**